src/test/java/PasswordGeneratorTest.java

```diff
@@ -1,18 +1,51 @@
-/*
- * This Java source file was generated by the Gradle 'init' task.
- */
 import org.junit.Test;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class PasswordGeneratorTest {
```

Comment:
We seem to be missing a few tests - we've checked the length and the number of numbers but what about the other conditions?